src/main/java/Controllers/ViewMainController.java

```diff
@@ -0,0 +1,201 @@
+package Controllers;
+
+import dbServicies.FruitDbService;
+import dbServicies.NoSqlFruitDbService;
+import dbServicies.SqlFruitDbService;
+import entitys.FruitRow;
+import initdb.InitNoSqlDb;
+import initdb.InitSqlDb;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.control.TextField;
+import javafx.scene.control.cell.PropertyValueFactory;
+import org.controlsfx.control.textfield.AutoCompletionBinding;
+import org.controlsfx.control.textfield.TextFields;
+import parameters.DbParameters;
+import parameters.TableParameters;
+import util.AlertWindow;
+import util.TextFieldsValidator;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ViewMainController {
+
+    private FruitDbService fruitDBService;
+
+    public static ObservableList<FruitRow> fruitRowData;
+
+    public static Set<String> fruitAutoComplete = new LinkedHashSet<String>();
+    public static Set<String> sortAutoComplete = new LinkedHashSet<String>();
+    public static Set<String> providerAutoComplete = new LinkedHashSet<String>();
+
```

Comment:
Ok. Imagine a create public class AutoCompleteHolder with fields, getters, setters and so on.
Then, to fill my fields (like fruitAutoComplete) i must create new AutoCompleteHolder and put it into variable like "autoCompleteHolder". So i can do it inside ViewMainController or SqlFruitDbService class:
1) i cant get ViewMainController instance because i never created it.
2) If i do it inside SqlFruitService class i cant reach my autoCompleteHolder variable inside ViewMainController.

So where should i create my AutoCompleteHolder variable?

Reply:
Upd. Done with singleton.